packages/fxa-auth-server/bin/key_server.js

```diff
@@ -69,130 +70,120 @@ function run(config) {
+    log,
+    serverPublicKeys,
+    signer,
+    database,
+    oauthdb,
+    senders.email,
+    senders.sms,
+    Password,
+    config,
+    customs,
+    zendeskClient,
+    subhub,
+    statsd,
+    profile,
+    stripeHelper
+  );
+
+  const server = await Server.create(
+    log,
+    error,
+    config,
+    routes,
+    database,
+    oauthdb,
+    translator
+  );
 
   function logStatInfo() {
     log.stat(server.stat());
     log.stat(Password.stat());
   }
+  const statsInterval = setInterval(logStatInfo, 15000);
 
-  const DB = require('../lib/db')(config, log, Token, UnblockCode);
+  try {
+    await server.start();
+    log.info({
+      op: 'server.start.1',
+      msg: `running on ${server.info.uri}`,
+    });
+  } catch (err) {
+    log.error({
+      op: 'server.start.1',
+      msg: 'failed startup with error',
+      err: { message: err.message },
+    });
+  }
 
-  return P.all([
-    DB.connect(config[config.db.backend]),
-    require('../lib/senders/translator')(
-      config.i18n.supportedLanguages,
-      config.i18n.defaultLanguage
-    ),
-  ])
-    .spread(
-      (db, translator) => {
-        database = db;
-        oauthdb = require('../lib/oauthdb')(log, config, statsd);
-        subhub = require('../lib/subhub/client').client(log, config, statsd);
-        profile = require('../lib/profile/client')(log, config, statsd);
-
-        return require('../lib/senders')(
-          log,
-          config,
-          error,
-          translator,
-          oauthdb,
-          statsd
-        ).then(result => {
-          senders = result;
-          customs = new Customs(config.customsUrl);
-          const routes = require('../lib/routes')(
-            log,
-            serverPublicKeys,
-            signer,
-            db,
-            oauthdb,
-            senders.email,
-            senders.sms,
-            Password,
-            config,
-            customs,
-            zendeskClient,
-            subhub,
-            statsd,
-            profile
-          );
-
-          statsInterval = setInterval(logStatInfo, 15000);
-
-          async function init() {
-            server = await Server.create(
-              log,
-              error,
-              config,
-              routes,
-              db,
-              oauthdb,
-              translator
-            );
-            try {
-              await server.start();
-              log.info({
-                op: 'server.start.1',
-                msg: `running on ${server.info.uri}`,
-              });
-            } catch (err) {
-              log.error({
-                op: 'server.start.1',
-                msg: 'failed startup with error',
-                err: { message: err.message },
-              });
-            }
+  return {
+    server,
+    log: log,
+    close() {
+      return new P(resolve => {
+        log.info({ op: 'shutdown' });
+        clearInterval(statsInterval);
+        server.stop().then(() => {
```

Comment:
why not `await` the remaining `then`s in this file?

Reply:
Awwww.... cause I didn't see that one, doh!